tests/test_components.py

```diff
@@ -538,114 +543,128 @@ def test_parse(self):
-        component.raw_data = {
-            "collection1": {
-                "pages": ["page1"],
-                "name": "Collection 1",
-                "user": "user1",
-                "workflow": "wf1",
-            }
-        }
-        with patch.object(component, "get_or_create") as get_or_create:
-            component.parse()
-            collection_data = {
-                "author": self.user,
-                "workflow": self.wf1,
-                "name": "Collection 1",
-            }
-            self.assertEqual(
-                component.data["collection1"], get_or_create(collection_data)
-            )
+        component.parse()
 
-    def test_get_or_create(self):
-        # test add
-        collection_data = {
-            "author": self.user,
-            "workflow": self.wf1,
-            "name": "Collection 1",
-        }
-        collection = self.component.get_or_create(collection_data)
-        self.assertEqual(collection.id, 1)
-        collection_data2 = {
-            "author": self.user,
-            "workflow": self.wf1,
-            "name": "Collection 2",
-        }
-        collection2 = self.component.get_or_create(collection_data2)
-        self.assertEqual(collection2.id, 2)
+        collection = ModerationCollection.objects.get()
+        self.assertEqual(collection.name, "Collection 1")
+        self.assertEqual(collection.author, self.user)
+        self.assertEqual(collection.workflow, self.wf1)
+        # TODO: Is it correct that all collections created with json
+        # will always be in COLLECTING state and have created/modified
+        # dates set to now?
+        from djangocms_moderation.constants import COLLECTING
+        self.assertEqual(collection.status, COLLECTING)
+        self.assertEqual(collection.date_created, now())
+        self.assertEqual(collection.date_modified, now())
+
+    def test_parse_does_not_create_collection_if_already_exists(self):
+        """If a ModerationCollection with that name exists, it should not change"""
+        from djangocms_moderation.constants import IN_REVIEW
+        with freeze_time("2010-10-10"):
+            existing_collection = ModerationCollectionFactory(
+                name="Collection 1", status=IN_REVIEW)
+        component = self._get_collections_obj()
 
-        # test get
-        collection3 = self.component.get_or_create(collection_data)
-        self.assertEqual(collection.id, 1)
+        component.parse()
+
+        collection = ModerationCollection.objects.get()
+        self.assertEqual(collection.pk, existing_collection.pk)
+        self.assertEqual(collection.author, existing_collection.author)
+        self.assertEqual(collection.workflow, existing_collection.workflow)
+        self.assertEqual(collection.status, existing_collection.status)
+        self.assertEqual(collection.date_created, existing_collection.date_created)
+        self.assertEqual(collection.date_created, existing_collection.date_modified)
```

Comment:
should be `collection.date_modified`, right?

Reply:
Good catch